src/core/AutoRest/Simplify/CSharpSimplifier.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// 
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoRest.Core;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Rest;
+using Microsoft.Rest.Azure;
+
+namespace AutoRest.Simplify
+{
+    public class CSharpSimplifier
+    {
+        private static MetadataReference mscorlib;
+
+        private static MetadataReference Mscorlib
+        {
+            get
+            {
+                if (mscorlib == null)
+                {
+                    mscorlib = MetadataReference.CreateFromFile(typeof(object).Assembly.Location);
+                }
+
+                return mscorlib;
+            }
+        }
+
+        public async Task Run()
+        {
+            var op = new AzureAsyncOperation();
+            var restOp = new RestException();
+
+            var files = Settings.Instance.FileSystem.GetFiles(Settings.Instance.OutputDirectory, "*.cs",
+                    SearchOption.AllDirectories).
+                ToDictionary(each => each, each => Settings.Instance.FileSystem.ReadFileAsText(each));
+
+            var projectId = ProjectId.CreateNewId();
+            var solution = new AdhocWorkspace().CurrentSolution
+                .AddProject(projectId, "MyProject", "MyProject", LanguageNames.CSharp)
+                .AddMetadataReference(projectId, Mscorlib)
+                .AddMetadataReference(projectId, AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
would `typeof(AzureAsyncOperation).Assembly` do the same? Same with `RestException` below